File: examples/ruby-rspec/steps/hello_var.steps.rb
# frozen_string_literal: true

require 'varar'

steps(greeting: '', result: 0) do
  stimulus('I greet {string}') { |_state, name| { greeting: "Hello, #{name}!" } }
  sensor('the greeting should be {string}') { |state, _expected| state[:greeting] }

  stimulus('expression `{int}+{int}`') { |_state, a, b| { result: a + b } }
  sensor('evaluate to `{int}`') { |state, _expected| state[:result] }
end

File: examples/typescript-vitest/steps/hello-var.steps.ts
import { steps } from '@varar/varar'

const { stimulus, sensor } = steps(() => ({ greeting: '', result: 0 }))

stimulus('I greet {string}', (_state, name) => ({ greeting: `Hello, ${name}!` }))

sensor('the greeting should be {string}', (state, _expected) => state.greeting)

stimulus('expression `{int}+{int}`', (_state, op1, op2) => ({ result: op1 + op2 }))

sensor('evaluate to `{int}`', (state, _count) => state.result)
